src/lib/api/SearchApi.ts

```diff
@@ -120,3 +120,18 @@ export const readFetcher = async (linkId: number) => {
     throw new Error(`cannot find work with linkId ${linkId}`);
   }
 };
+
+export const fulfillFetcher = async (fulfillUrl, nyplIdentityCookie) => {
+  const url = new URL(fulfillUrl);
+  const res = await fetch(url.toString(), {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${nyplIdentityCookie.access_token}`,
+    },
+  });
+  if (res.ok) {
+    return res.url;
+  } else {
+    throw new Error(`Unable to download UP PDF`);
+  }
```

Comment:
I'm thinking this might be a good opportunity to sync with Apoorva on the potential errors that could come up here, based on what the server responds with, and checking to see if she has input on if we want to show anything to the user.

Based on her input, we should write a test to make sure the error gets shown in the way we expect it to.

Reply:
Working with Apoorva to get designs approved for a modal which would appear for servers errors unrelated to authenication. For authentication errors, it will redirect to the login page.